src/DataProtection/DataProtection/src/Repositories/DefaultKeyStorageDirectories.cs

```diff
@@ -49,14 +48,13 @@ private DefaultKeyStorageDirectories()
         {
             retVal = GetKeyStorageDirectoryFromBaseAppDataPath(localAppDataFromEnvPath);
         }
-        else if (userProfilePath != null)
+        else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && homePath != null)
         {
-            retVal = GetKeyStorageDirectoryFromBaseAppDataPath(Path.Combine(userProfilePath, "AppData", "Local"));
+            retVal = GetKeyStorageDirectoryFromBaseAppDataPath(Path.Combine(homePath, "AppData", "Local"));
```

Comment:
Is this any different from using Environment.SpecialFolder.LocalApplicationData at this point?

Reply:
I have changed it to `Environment.SpecialFolder.LocalApplicationData`.

Reply:
But `Environment.SpecialFolder.LocalApplicationData` is already checked first above so I don't think this makes sense.